src/components/Appointments/AppointmentDetails.vue

```diff
@@ -124,14 +131,18 @@ export default {
 		},
 	},
 	methods: {
-		save() {
-			this.$emit('save', {
+		async save() {
+			this.canClose = false
+			this.isLoading = true
+			await this.$emit('save', {
 				slot: this.timeSlot,
 				description: this.description,
 				email: this.email,
 				displayName: this.displayName,
 				timeZone: this.timeZone,
 			})
+			this.canClose = true
+			this.isLoading = false
```

Comment:
tip: wrap in a try-finally where you reset the two flags in the finally block. that ensures the state is also reset for errors.

Reply:
Nice tip, thanks! I have updated the code 👍